src/firecrest/config.py

```diff
@@ -182,42 +309,108 @@ class HPCCluster(CamelModel):
+    )
+    file_systems: List[FileSystem] = Field(
+        default_factory=list,
+        description="List of mounted file systems on the cluster, such as scratch or home directories.",
+    )
+    datatransfer_jobs_directives: List[str] = Field(
+        default_factory=list,
+        description="Custom scheduler flags passed to data transfer jobs (e.g. `-pxfer` for a dedicated partition).",
+    )
 
 
 class OpenFGA(CamelModel):
-    url: str
-    timeout: Optional[int] = 1
-    max_connections: Optional[int] = 100
+    """Authorization settings using OpenFGA."""
+
+    url: str = Field(..., description="OpenFGA API base URL.")
+    timeout: Optional[int] = Field(
+        1,
+        description="Connection timeout in seconds. When `None` the timeout is disabled.",
+    )
+    max_connections: int = Field(
+        100,
+        description="Max HTTP connections per host. When set to `0`, there is no limit.",
+    )
 
 
 class SSHKeysService(CamelModel):
-    url: str
-    max_connections: Optional[int] = 100
+    """External service for managing SSH keys."""
+
+    url: str = Field(..., description="URL of the SSH keys management service.")
+    max_connections: int = Field(
+        100,
+        description=(
+            "Maximum concurrent connections to the service. When set to "
+            "`0`, there is no limit."
+        ),
+    )
 
 
 class Auth(CamelModel):
-    authentication: Oidc = None
-    authorization: Optional[OpenFGA] = None
+    """Authentication and authorization configuration."""
+
+    authentication: Oidc = Field(
+        ...,
+        description=(
+            "OIDC authentication settings. More info in the "
+            "[authentication section](../arch/auth/README.md#authentication)."
+        ),
+    )
+    authorization: Optional[OpenFGA] = Field(
+        None,
+        description=(
+            "Authorization settings via OpenFGA. More info in [the "
+            "authorization section](../arch/auth/README.md#authorization)."
+        ),
+    )
 
 
 class Settings(BaseSettings):
-    # FastAPI App variables
-    app_debug: bool = False
-    app_version: str = "2.x.x"
-    apis_root_path: str = ""
-    doc_servers: Optional[List[Dict]] = None
-    # Authentication and Authorization settings
-    auth: Auth = None
-    # SSH Credentials
-    ssh_credentials: SSHKeysService | Dict[str, SSHUserKeys]
-    # HPC Clusters definition
-    clusters: List[HPCCluster] = []
-    # HPC Storage definition
-    storage: Optional[Storage] = None
+    """FirecREST configuration. Loaded from a YAML file."""
+
+    app_debug: bool = Field(
+        False, description="Enable debug mode for the FastAPI application."
+    )
+    app_version: str = Field("2.x.x", description="Application version string.")
+    apis_root_path: str = Field(
+        "",
+        description="Base path prefix for exposing the APIs.",
+    )
+    doc_servers: Optional[List[dict]] = Field(
+        None,
```

Comment:
I've noticed that there are some parts of the doc where the `Default` and the `Description` columns values are exchanged. For what I see those with type `None` are the ones that present this issue.

Reply:
Good catch, thanks!! I think I fixed it now